eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/BackPropagatedRSWriter.java

```diff
@@ -0,0 +1,259 @@
+                    bulkSVOEInserts = dbsnpSVOEInserts;
+                    numDbsnpRSAssignments += 1;
+                }
+                ClusteredVariantEntity newRSRecordForOriginalAssembly =
+                        getSuitableRSFromRemappedSS(ssInRemappedAssemblyGroupedByID, submittedVariantEntity);
+                Long rsToAssign = newRSRecordForOriginalAssembly.getAccession();
+
+                if (clusteringWriter.getClusteredVariantCollection(rsToAssign).equals(ClusteredVariantEntity.class)) {
+                    bulkCVEInserts = evaCVEInserts;
+                    numEVARSInserts += 1;
+                } else {
+                    bulkCVEInserts = dbsnpCVEInserts;
+                    numDbsnpRSInserts += 1;
+                }
+                bulkCVEInserts.insert(newRSRecordForOriginalAssembly);
+                clusteringCounts.addClusteredVariantsCreated(1);
+
+                Query queryToLookUpSSHash = query(where(ID_ATTRIBUTE).is(submittedVariantEntity.getHashedMessage()));
+                Update updateRSInOriginalAssembly = Update.update(RS_ATTRIBUTE, rsToAssign);
+                bulkSVEUpdates.updateOne(queryToLookUpSSHash, updateRSInOriginalAssembly);
+                clusteringCounts.addSubmittedVariantsClustered(1);
+
+                bulkSVOEInserts.insert(getUpdateOperation(submittedVariantEntity, rsToAssign));
+                clusteringCounts.addSubmittedVariantsUpdateOperationWritten(1);
+            }
+        }
+
+        handleClusteredVariantInserts(evaCVEInserts, dbsnpCVEInserts, numEVARSInserts, numDbsnpRSInserts);
+
+        if (numEvaRSAssignments > 0) {
+            evaSVEUpdates.execute();
+            evaSVOEInserts.execute();
+        }
+
+        if (numDbsnpRSAssignments > 0) {
+            dbsnpSVEUpdates.execute();
+            dbsnpSVOEInserts.execute();
+        }
+    }
+
+    private void handleClusteredVariantInserts(BulkOperations evaCVEInserts, BulkOperations dbsnpCVEInserts,
+                                               int numEVARSInserts, int numDbsnpRSInserts) {
+        try {
+            if (numEVARSInserts > 0) {
+                evaCVEInserts.execute();
+            }
+
+            if (numDbsnpRSInserts > 0) {
+                dbsnpCVEInserts.execute();
+            }
+        }
+        catch (DuplicateKeyException duplicateKeyException) {
+            MongoBulkWriteException writeException = ((MongoBulkWriteException) duplicateKeyException.getCause());
+            // Consider a scenario where we are back-propagating rs ID:
+            // RS1 corresponding to SS1 from ASM2 (remapped assembly) to ASM1 (original assembly)
+            // If the hash for back-propagated RS1 already exists in ASM1,
+            // it means that, for some reason, SS1 was not clustered with RS1 when ASM1 underwent clustering.
+            // It is better to flag such issues for later analysis.
+            extractUniqueHashesForDuplicateKeyError(writeException)
```

Comment:
Actually this scenario seem perfectly possible:
 1. Clustering happens on ASM1 and create RS1 from SS0
 2. SS1 is added to ASM1 (but not clustered)
 3. SS1 (and SS0) is remapped to ASM2 (called rSS1 and rSS0)
 4. RS1 is created in ASM2 and assigned to rSS1
 5. Back-propagate to ASM1 and assign RS1 to SS1
 6. Create RS1 in ASM1 and hit duplicate 

This might be a good reason not to create the ClusteredVariant in the original assembly when back propagating.

Reply:
Fair point regarding this situation happening in a partially clustered assembly. It should be noted that the RS assignment will still happen even if the RS hash collision is found and this error message is printed. I can downgrade this message to a warning.